plugin/plugins/telemetry/telemetry.go

```diff
@@ -0,0 +1,538 @@
+package telemetry
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"path"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+	logging "github.com/ipfs/go-log/v2"
+	ipfs "github.com/ipfs/kubo"
+	"github.com/ipfs/kubo/config"
+	"github.com/ipfs/kubo/core"
+	"github.com/ipfs/kubo/core/corerepo"
+	"github.com/ipfs/kubo/plugin"
+	"github.com/libp2p/go-libp2p/core/network"
+	"github.com/libp2p/go-libp2p/core/pnet"
+	multiaddr "github.com/multiformats/go-multiaddr"
+	manet "github.com/multiformats/go-multiaddr/net"
+)
+
+var log = logging.Logger("telemetry")
+
+const (
+	modeEnvVar   = "IPFS_TELEMETRY_PLUGIN_MODE"
+	uuidFilename = "telemetry_uuid"
+	endpoint     = "https://telemetry.ipshipyard.dev"
+	sendDelay    = 15 * time.Minute
+)
+
+type pluginMode int
+
+const (
+	modeInfo pluginMode = iota
+	modeOptIn
+	modeOptOut
+)
+
+// repoSizeBuckets defines size thresholds for categorizing repository sizes.
+// Each value represents the upper limit of a bucket in bytes (except the last)
+var repoSizeBuckets = []uint64{
+	1 << 30,   // 1 GB
+	5 << 30,   // 5 GB
+	10 << 30,  // 10 GB
+	100 << 30, // 100 GB
+	500 << 30, // 500 GB
+	1 << 40,   // 1 TB
+	10 << 40,  // 10 TB
+	11 << 40,  // + anything more than 10TB falls here.
+}
+
+var uptimeBuckets = []time.Duration{
+	1 * 24 * time.Hour,
+	24 * time.Hour,
```

Comment:
Same as previous bucket
```suggestion
	2 * 24 * time.Hour,
```